### ISSUE TITLE
Setting _publicationState to live does not work with count API method on MongoDB

### ISSUE THREAD

**Issue by gondar00** [strapi/strapi#9092]
### Describe the bug

Trying to filter the `{api}/count` using `_publicationState=live` returns me the total count (draft+published)

### Steps to reproduce the behavior

Hit the endpoint with _publicationState=live query param

### Expected behavior

I was expecting only the count of the published entities

**Comment by gondar00** [comment]
workaround is to add the query param `published_at_null=false` to retrieve only published entries.. 

But the response time takes a hit

**Comment by Convly** [comment]
Hi, thanks for raising your issue.

I tried to reproduce this behaviour in a brand new app but didn't succeed to do so.

![image](https://user-images.githubusercontent.com/25851739/104187679-72fc0780-5418-11eb-974a-00ecf1dabb93.png)
![image](https://user-images.githubusercontent.com/25851739/104187814-a2127900-5418-11eb-87bc-5a4af02f8f51.png)


Could you give more information about your configuration (such as your strapi version, node version, and database used)?
Also, if possible could you provide to me an example project with the issue or a list of steps to set up a project like you did?

Finally, did you checked your content-type had the draft & publish feature activated?

Thanks! :slightly_smiling_face:

**Comment by Convly** [comment]
Hi, thanks for raising your issue.

I tried to reproduce this behaviour in a brand new app but didn't succeed to do so.

![image](https://user-images.githubusercontent.com/25851739/104187950-cff7bd80-5418-11eb-92d9-9e0b449bca9e.png)

![image](https://user-images.githubusercontent.com/25851739/104187814-a2127900-5418-11eb-87bc-5a4af02f8f51.png)


Could you give more information about your configuration (such as your strapi version, node version, and database used)?
Also, if possible could you provide to me an example project with the issue or a list of steps to set up a project like you did?

Finally, did you checked your content-type had the draft & publish feature activated?

Thanks! :slightly_smiling_face:

**Comment by gondar00** [comment]
Hi @Convly! Thanks for your quick response 🥇 

I have used a custom controller function where I have overridden the `find` function and I want to append the count also along with the response object, code snippet as below - 

```
  async find(ctx) {
    let entities;
    let count;

    ctx.query = {
      ...ctx.query,
      _publicationState: 'live'
    }

    if (ctx.query._q) {
      entities = await strapi.services.article.search(ctx.query);
      count = await strapi.services.article.countSearch(ctx.query);
    } else {
      entities = await strapi.services.article.find(ctx.query);
      count = await strapi.services.article.count(ctx.query);
    }

    const payload = {
      entities,
      count,
      model: strapi.models.article,
    };

    return customResponse(payload);
  }
```

| strapi version | 3.4.1 |
| node version | 12.18.4|
| database | mongodb |
| draft & publish feature activated| true|

**Comment by Convly** [comment]
I was able to reproduce the issue and it seems linked to MongoDB, I'll look into it and keep you posted.
Thanks again for reporting this.

**Comment by Convly** [comment]
The bug has been fixed in #9094. It'll be available in 3.4.2.

Also, a little tip: in your code above, you don't need to explicitly set the `publicationState` to `live` in your context since it'll be automatically added in the core-api services (when you call `strapi.services.article.find/count`)
https://github.com/strapi/strapi/blob/0f5a93268af2a8015601822ad131da640a0bfd54/packages/strapi/lib/core-api/service.js#L166-L175
The `getFetchParams` adds a default `publicationState: live` if not provided
